editor/components/warning/style.scss

```diff
@@ -1,27 +1,53 @@
-.editor-warning {
+.editor-block-list__block .editor-warning {
```

Comment:
I mentioned at https://github.com/WordPress/gutenberg/pull/7886#discussion_r202762204 that this component is used for more than just the warning for the single block in a list. It's also used for the top-level error handler, which this change broke. See #8677.